compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -402,6 +402,20 @@ public boolean containsErrorType(BType bType) {
         return type.tag == TypeTags.ERROR;
     }
 
+    public boolean containsNilType(BType bType) {
+        BType type = getReferredType(bType);
+        if (type.tag == TypeTags.UNION) {
+            return ((BUnionType) type).getMemberTypes().stream()
```

Comment:
Should we avoid using streams? :thinking:

Reply:
Updated